docassemble/AssemblyLine/al_document.py

```diff
@@ -396,13 +396,82 @@ class ALDocument(DADict):
   Required attributes:
     - filename: name used for output PDF
     - title: display name for the output PDF
-    - enabled
-    - has_addendum: The default value is False. Set to True if the document has overflow, like for a PDF template.
+    - enabled: if this document should be created. See examples.
 
   Optional attribute:
     - addendum: an attachment block
-    - overflow_fields
-
+    - overflow_fields: ALAddendumFieldDict instance. These values will be used to detect and handle overflow.
+    - has_addendum: Defaults to False. Set to True if the document could have overflow, like for a PDF template.
+  
+  Examples:
+  
+  Simple use where the document is always enabled and will have no addendum
+  ```
+  ---
+  objects:
+    - my_doc: ALDocument.using(filename="myDoc.pdf", title="myDoc", enabled=True)
+  --- 
+  attachment:
+      variable name: my_doc[i]  # This will usually be "final" or "preview"
+      ...
+  ```
+  
+  Enable a document conditionally
```

Comment:
Extremely minor: the addendum examples are a lot simpler than the conditionally enabled example. It might help readability if the addendum examples come before this enabling example.

Reply:
Note that it is actually not possible to conditionally enable the addendum. You must specify it at initialization after the recent change.

Addendum is on/off at a document level, but is only triggered if the document has overflow. I can commit an edit if that is preferred.

Reply:
Clearly good evidence of why we need this documentation, so thank you @plocket